src/collector.ts

```diff
@@ -125,14 +126,16 @@ class NaiveGomodParser {
                         const entry: IKeyValueEntry = new KeyValueEntry(pkgName, { line: 0, column: 0 });
                         entry.value = new Variant(ValueType.String, 'v' + version[0]);
                         entry.value_position = { line: index + 1, column: version.index };
-                        dependencies.push(new Dependency(entry));
+                        const dep = new Dependency(entry);
+                        dependencies.push(dep);
+                        // Push all direct and indirect modules present in go.mod (manifest) 
+                        finalDeps.push(dep);
```

Comment:
Why one more array? Why not just return `{goModDeps ⋃ {goModDeps ⋂ goImportPackages}}`?

Reply:
This is taken care in new code.